File: frontend/packages/loop-base/api-schema/src/api/idl/data/domain/dataset_job.ts
// Copyright (c) 2025 coze-dev Authors
// SPDX-License-Identifier: Apache-2.0
import * as dataset from './dataset';
export { dataset };
/** 通用任务类型 */
export enum JobType {
  ImportFromFile = 1,
  ExportToFile = 2,
  ExportToDataset = 3,
}
/** 通用任务状态 */
export enum JobStatus {
  Undefined = 0,
  /** 待处理 */
  Pending = 1,
  /** 处理中 */
  Running = 2,
  /** 已完成 */
  Completed = 3,
  /** 失败 */
  Failed = 4,
  /** 已取消 */
  Cancelled = 5,
}
export const LogLevelInfo = "info";
export const LogLevelError = "error";
export const LogLevelWarning = "warning";
/** 通用任务日志 */
export interface JobLog {
  content: string,
  level: string,
  timestamp: string,
  hidden: boolean,
}
export enum FileFormat {
  JSONL = 1,
  Parquet = 2,
  CSV = 3,
  XLSX = 4,
  /** [100, 200) 压缩格 */
  ZIP = 100,
}
export enum SourceType {
  File = 1,
  Dataset = 2,
}
export interface DatasetIOFile {
  provider: dataset.StorageProvider,
  path: string,
  /** 数据文件的格式 */
  format?: FileFormat,
  /** 压缩包格式 */
  compress_format?: FileFormat,
  /** path 为文件夹或压缩包时，数据文件列表, 服务端设置 */
  files?: string[],
  /** 原始的文件名，创建文件时由前端写入。为空则与 path 保持一致 */
  original_file_name?: string,
  /** 文件下载地址 */
  download_url?: string,
  /** 存储提供方ID，目前主要在 provider==imagex 时生效 */
  provider_id?: string,
  /** 存储提供方鉴权信息，目前主要在 provider==imagex 时生效 */
  provider_auth?: ProviderAuth,
}
export interface ProviderAuth {
  /** provider == VETOS 时，此处存储的是用户在 fornax 上托管的方舟账号的ID */
  provider_account_id?: string
}
export interface DatasetIODataset {
  space_id?: string,
  dataset_id: string,
  version_id?: string,
}
export interface DatasetIOEndpoint {
  file?: DatasetIOFile,
  dataset?: DatasetIODataset,
}
/** DatasetIOJob 数据集导入导出任务 */
export interface DatasetIOJob {
  id: string,
  app_id?: number,
  space_id: string,
  /** 导入导出到文件时，为数据集 ID；数据集间转移时，为目标数据集 ID */
  dataset_id: string,
  job_type: JobType,
  source: DatasetIOEndpoint,
  target: DatasetIOEndpoint,
  /** 字段映射 */
  field_mappings?: FieldMapping[],
  option?: DatasetIOJobOption,
  /** 运行数据, [20, 100) */
  status?: JobStatus,
  progress?: DatasetIOJobProgress,
  errors?: dataset.ItemErrorGroup[],
  /** 通用信息 */
  created_by?: string,
  created_at?: string,
  updated_by?: string,
  updated_at?: string,
  started_at?: string,
  ended_at?: string,
}
export interface DatasetIOJobOption {
  /** 覆盖数据集 */
  overwrite_dataset?: boolean
}
export interface DatasetIOJobProgress {
  /** 总量 */
  total?: string,
  /** 已处理数量 */
  processed?: string,
  /** 已成功处理的数量 */
  added?: string,
  /**
   * 子任
   * 可空, 表示子任务的名称
  */
  name?: string,
  /** 子任务的进度 */
  sub_progresses?: DatasetIOJobProgress[],
}
export interface FieldMapping {
  source: string,
  target: string,
}

File: backend/modules/evaluation/domain/entity/dataset_job.go
// Copyright (c) 2025 coze-dev Authors
// SPDX-License-Identifier: Apache-2.0

package entity

// JobType 通用任务类型
type JobType int64

const (
	JobType_ImportFromFile  JobType = 1
	JobType_ExportToFile    JobType = 2
	JobType_ExportToDataset JobType = 3
)

// JobStatus 通用任务状态
type JobStatus int64

const (
	JobStatus_Undefined JobStatus = 0
	// 待处理
	JobStatus_Pending JobStatus = 1
	// 处理中
	JobStatus_Running JobStatus = 2
	// 已完成
	JobStatus_Completed JobStatus = 3
	// 失败
	JobStatus_Failed JobStatus = 4
	// 已取消
	JobStatus_Cancelled JobStatus = 5
)

type DatasetIOJobOption struct {
	// 覆盖数据集
	OverwriteDataset  *bool               `json:"overwrite_dataset,omitempty"`
	FieldWriteOptions []*FieldWriteOption `json:"field_write_options,omitempty"`
}

type DatasetIOJobProgress struct {
	// 总量
	Total *int64 `json:"total,omitempty"`
	// 已处理数量
	Processed *int64 `json:"processed,omitempty"`
	// 已成功处理的数量
	Added *int64 `json:"added,omitempty"`
	/*子任务*/
	Name *string `json:"name,omitempty"`
	// 子任务的进度
	SubProgresses []*DatasetIOJobProgress `json:"sub_progresses,omitempty"`
}

// DatasetIOJob 数据集导入导出任务
type DatasetIOJob struct {
	ID      int64  `json:"id"`
	AppID   *int32 `json:"app_id,omitempty"`
	SpaceID int64  `json:"space_id"`
	// 导入导出到文件时，为数据集 ID；数据集间转移时，为目标数据集 ID
	DatasetID int64              `json:"dataset_id"`
	JobType   JobType            `json:"job_type"`
	Source    *DatasetIOEndpoint `json:"source"`
	Target    *DatasetIOEndpoint `json:"target"`
	// 字段映射
	FieldMappings []*FieldMapping     `json:"field_mappings,omitempty"`
	Option        *DatasetIOJobOption `json:"option,omitempty"`
	/* 运行数据, [20, 100) */
	Status   *JobStatus            `json:"status,omitempty"`
	Progress *DatasetIOJobProgress `json:"progress,omitempty"`
	Errors   []*ItemErrorGroup     `json:"errors,omitempty"`
	/* 通用信息 */
	CreatedBy *string `json:"created_by,omitempty"`
	CreatedAt *int64  `json:"created_at,omitempty"`
	UpdatedBy *string `json:"updated_by,omitempty"`
	UpdatedAt *int64  `json:"updated_at,omitempty"`
	StartedAt *int64  `json:"started_at,omitempty"`
	EndedAt   *int64  `json:"ended_at,omitempty"`
}

type ImportEvaluationSetParam struct {
	WorkspaceID     int64
	EvaluationSetID int64
	File            *DatasetIOFile
	FieldMappings   []*FieldMapping
	Option          *DatasetIOJobOption
}
